app/modules/code_provider/github/github_service.py

```
@@ -506,22 +506,51 @@ async def get_branch_list(self, repo_name: str):
         except Exception as e:
             logger.error(
-                f"Error fetching branches for repo {repo_name}: {str(e)}", exc_info=True
+                f"Unexpected error fetching branches for repo {repo_name}: {str(e)}", 
+                exc_info=True
             )
             raise HTTPException(
-                status_code=404,
-                detail=f"Repository not found or error fetching branches: {str(e)}",
+                status_code=500,
+                detail=f"Unexpected error: {str(e)}"
             )
```

Ruff (0.8.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)